dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java

```diff
@@ -174,134 +174,143 @@ public ProcessingEnvironment getEnv() {
         return env;
     }
 
-    /**
-     * Perform the actual analysis. Should only be called once after the object has been created.
-     */
+    /** Perform the actual analysis. */
     public void performAnalysis(ControlFlowGraph cfg) {
         assert !isRunning;
         isRunning = true;
 
-        init(cfg);
-
-        while (!worklist.isEmpty()) {
-            Block b = worklist.poll();
-
-            switch (b.getType()) {
-                case REGULAR_BLOCK:
-                    {
-                        RegularBlock rb = (RegularBlock) b;
-
-                        // apply transfer function to contents
-                        TransferInput<A, S> inputBefore = getInputBefore(rb);
-                        currentInput = inputBefore.copy();
-                        TransferResult<A, S> transferResult = null;
-                        Node lastNode = null;
-                        boolean addToWorklistAgain = false;
-                        for (Node n : rb.getContents()) {
-                            transferResult = callTransferFunction(n, currentInput);
-                            addToWorklistAgain |= updateNodeValues(n, transferResult);
-                            currentInput = new TransferInput<>(n, this, transferResult);
-                            lastNode = n;
-                        }
-                        // loop will run at least once, making transferResult non-null
-
-                        // propagate store to successors
-                        Block succ = rb.getSuccessor();
-                        assert succ != null
-                                : "regular basic block without non-exceptional successor unexpected";
-                        propagateStoresTo(
-                                succ, lastNode, currentInput, rb.getFlowRule(), addToWorklistAgain);
-                        break;
-                    }
+        try {
+            init(cfg);
+
+            while (!worklist.isEmpty()) {
+                Block b = worklist.poll();
+
+                switch (b.getType()) {
```

Comment:
How about putting this whole switch block into a protected helper method, something like `analyseBlock`? That makes the overall control flow much easier to read.